### PR TITLE
osm widget

### LINKED ISSUE #126: [job] GPS widget plotting GPS trajectory on top of maps
Is anybody working (or planning to work) on a GPS widget which plots e.g. long / lat trajectories on top of e.g. Open Street Map imagery? We would love to have such a widget and we would also be happy to contribute an implementation of it if somebody can give us some pointers on how to add new widgets to the bagbunker. Cheers!

### PR DESCRIPTION
**Pull request by chaoflow**
Please adjust https://github.com/bosch-ros-pkg/bagbunker/blob/osm/src/deepfield_jobs/deepfield_jobs/osm.py to your own needs before merging into develop.

In order to test, please build your own docker image from the osm branch or use docker.ternaris.com/bagbunker/bagbunker:develop and follow the newly added instructions for [frontend development](https://github.com/bosch-ros-pkg/bagbunker/tree/develop#frontend-development)

Cc: @NikolausDemmel @dejanpan @durko 

closes #126

### CODE REVIEW

File: src/marv/marv/frontend/app/pods/widget-osm/component.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    boot: Ember.on('didInsertElement', function() {
+        const map = L.map(this.$().find('.map')[0], {
+            minZoom: 0,
+            maxZoom: 20,
+            attributionControl: false,
+            maxBounds: L.LatLngBounds(L.latLng([75,-180]), L.latLng([-75, 180]))
+        });
+        this.set('map', map);
+
+        L.tileLayer(`${window.location.protocol}//{s}.osm.ternaris.com/mapbox-studio-osm-bright/{z}/{x}/{y}${L.Browser.retina ? '@2x' : ''}.png`, {
```
Comment (dejanpan) [inline src/marv/marv/frontend/app/pods/widget-osm/component.js]: could you explain what this line does? Would it still work  if e.g. ternaris.com is down?

Comment (chaoflow) [inline src/marv/marv/frontend/app/pods/widget-osm/component.js]: That configures our tile server to be used, which we intend to provide for bagbunker installations.

**Comment by MatthiasRoland** [comment]
Awesome, works like a charm and looks really great :+1: 

I changed the color code based on the default http://docs.ros.org/api/sensor_msgs/html/msg/NavSatStatus.html message definition. Looks good for our datasets and I hope it works for others as well.

**Comment by NikolausDemmel** [comment]
@dejanpan, how do you handle who merges PRs at the moment? Should we merge it when it works for us, should you or someone else do it?

**Comment by dejanpan** [comment]
Assignee normally tests it and merges it. Go ahead and do it if it works
for you.

**Comment by MatthiasRoland** [comment]
* tested on several bags
* implemented separate osm job for our custom GPS messages

Everything works fine :) Thanks @chaoflow !

**Comment by dejanpan** [comment]
👍 I then also delete the branch normally so the repo is not too polluted.

**Comment by NikolausDemmel** [comment]
Very nice! Thanks @chaoflow !

**Pull request merged**